wagtail/core/query.py

```diff
@@ -237,6 +272,71 @@ def not_public(self):
         """
         return self.exclude(self.public_q())
 
+    def _get_closest_view_restriction(self, page):
+        """
+        Return the PageViewRestriction most applicable to the provided ``page`` instance.
+
+        View restrictions apply to the selected page AND their descendants, so for each
+        page, it's the restriction applied to the 'closest' ancestor that determines
+        what restrictions apply to it.
+        """
+        for vr in self.all_view_restrictions:
+            if page.id == vr.page.id or page.is_descendant_of(vr.page):
+                return vr
+
+    def for_user(self, user=None, include_password_protected=False):
```

Comment:
I'm confused about what `include_password_protected` means. Does it unlock all the password-restricted pages?

Reply:
the docs/code further down means it includes them because you can't know whether the user knows the password or not

Reply:
@kaedroho Say you wanted to filter search results on a site to make them 'applicable' to a user/request - By default, I exclude any pages with a password restriction. However, it might be that the developer is okay with password-protected pages being included in results, and the user getting the 'Enter the password' prompt if they click on the result and are taken to the page.